DataBiosphere/Saturn-UI
# IA-4065: Add Hail Batch as a launchable app, behind feature preview

### Jira Ticket: https://broadworkbench.atlassian.net/browse/IA-4065

This PR adds Hail Batch as a new app type for Azure workspaces. It is currently controlled by a feature flag. The backend functionality is already merged.




## Summary of changes:


### What
I made some inline PR comments about specific decisions, etc. Screenshots of overall flow below.

Feature flag: 
<img width="1318" alt="image" src="https://github.com/DataBiosphere/terra-ui/assets/5368863/89c89c17-9f29-418c-bb2d-efc792539d37">

App is creatable from Cloud Env modal:
<img width="421" alt="image" src="https://github.com/DataBiosphere/terra-ui/assets/5368863/62d2528c-e378-4c32-a70d-da16b7f52c71">

Running app listed in context bar:
<img width="106" alt="image" src="https://user-images.githubusercontent.com/5368863/236546823-1c8cadc5-8930-4ed8-9aab-055287caf2a0.png">

Running app can be opened or deleted:
<img width="680" alt="image" src="https://user-images.githubusercontent.com/5368863/236544403-e89f6505-0bec-47ab-98a8-4d0ec104ab21.png">

App UI is proxied:
<img width="1188" alt="image" src="https://user-images.githubusercontent.com/5368863/236544525-e253af25-bb12-48d4-80e9-114bcf39d655.png">


### Why
- Hail Batch is a Broad-developed analysis tool which will provide value to users.
- We've support Hail _Query_ for a long time on GCP; Hail Batch is the evolution of the project, and being supported on Azure for the first time.

### Testing strategy
- Enable feature flag; create/list/open/delete Hail Batch app from the UI
   - If interested in actually submitting jobs to Hail Batch, ping Rob and I can share some notebook code
- Verify functionality is not exposed if feature flag is disabled
